evm/src/assets/sharedComponents/AccessControl.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.25;
+
+import { BytesParsing } from "wormhole-sdk/libraries/BytesParsing.sol";
+import "./ids.sol";
+
+//rationale for different roles (owner, admin):
+// * owner should be a mulit-sig / ultra cold wallet that is only activated in exceptional
+//     circumstances.
+// * admin should also be either a cold wallet or Admin contract. In either case,
+//     the expectation is that multiple, slightly less trustworthy parties than the owner will
+//     have access to it, lowering trust assumptions and increasing attack surface. Admins
+//     perform rare but not exceptional operations.
+
+struct AccessControlState {
+  address   owner; //puts owner address in eip1967 admin slot
+  address   pendingOwner;
+  address[] admins;
+  mapping(address => bool) isAdmin;
```

Comment:
This should be a mapping to the index in the admins array (+1 so 0 can mean "does not exist"). That way we can also skip the linear search when updating.

Reply:
Hmm, I'd rather provide both so that admin checks are still cheap when needed.

Reply:
`mapping(address => uint256)` will serve both purposes. A non-zero value means the address is an admin and can be found at index `isAdmin[addr] - 1` of `admins`.